atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+                    elif len(self._system_info[sec_key][box_key]) != n_frames:  # TODO Should really check that the stored times for these quantities are exact, not just the same length
+                        self.logger.warning('Distinct trajectory lengths of box vectors and positions is not supported. ' + box_key + ' values will not be stored.')
+                        continue
+                else:
+                    self._system_info[sec_key][box_key] = [self._system_info[sec_key][box_key]] * n_frames
+        return self._system_info
+
+    @property
+    def observable_info(self):
+        if self._observable_info is None:
+            observables_group = self.h5md_groups.get('observables')
+            if observables_group is None:
+                return
+
+            self._observable_info = {
+                'configurational': {},
+                'ensemble_average': {},
+                'correlation_function': {}
+            }
+
+            def get_observable_paths(observable_group, current_path, paths):
+                for obs_key in observable_group.keys():
+                    path = obs_key + '.'
+                    observable = self._data_parser.hdf5_getter(observable_group, obs_key)
+                    observable_type = self._data_parser.hdf5_getter(observable_group, obs_key).attrs.get('type')
+                    if not observable_type:
+                        paths = get_observable_paths(observable, current_path + path, paths)
+                    else:
+                        paths.append(current_path + path[:-1])
+
+                return paths
+
+            observable_paths = get_observable_paths(observables_group, current_path='', paths=[])
+
+            for path in observable_paths:
+                observable = self._data_parser.hdf5_getter(observables_group, path)
+                observable_type = self._data_parser.hdf5_getter(observables_group, path).attrs.get('type')
+                observable_name = path.split('.')[0]
+                observable_label = '-'.join(path.split('.')[1:]) if len(path.split('.')) > 1 else ''
+                if observable_name not in self._observable_info[observable_type].keys():
+                    self._observable_info[observable_type][observable_name] = {}
+                self._observable_info[observable_type][observable_name][observable_label] = {}
+                for key in observable.keys():
+                    observable_attribute = self._data_parser.hdf5_getter(observable, key)
+                    if type(observable_attribute) == h5py.Group:
+                        self.logger.warning('Group structures within individual observables not supported. ' + key + ' values will not be stored.')
+                        continue
+                    self._observable_info[observable_type][observable_name][observable_label][key] = observable_attribute
+        return self._observable_info
+
+    def get_atomsgroup_fromh5md(self, nomad_sec, h5md_sec_particlesgroup):
```

Comment:
this does seem to be a get function but parse_atomsgroup?

Reply:
indeed, done.